### ISSUE TITLE
Triton Backend Trains Slower Than Cuda?

### ISSUE THREAD

**Issue by johnrobinsn** [johnsmith0031/alpaca_lora_4bit#66]
More curious about what others are seeing....

I'm fine-tuning llama30b on a 24G titan RTX card and using same everything (model, dataset (alpaca), hyperparameters, gradient checkpointing on) and just trying out fine-tuning with different the two backends (cuda vs triton).

Cuda is a little more than 3s/it  and Triton is a little under 4s/it... I guess I was hoping/expecting triton to perform a bit better (but at least no worse...)

What are others seeing?

Thanks!

**Comment by Ph0rk0z** [comment]
New cuda is slow AF too. I loaded the same exact GPTQv1 llama-30b 4bit model. Quadro P6000

Old implementation:
```
Output generated in 6.92 seconds (4.19 tokens/s, 29 tokens, context 382)
Output generated in 8.72 seconds (5.05 tokens/s, 44 tokens, context 382)
```
New Cuda
```
Output generated in 26.57 seconds (1.09 tokens/s, 29 tokens, context 376)
Output generated in 26.02 seconds (1.11 tokens/s, 29 tokens, context 376)

```

I can't get any model to load with Triton because all of what I downloaded had no group size, just the new format. I am looking for one to try or I'll encode one myself.

I'm assuming training will follow the same story.

**Comment by johnrobinsn** [comment]
Looks like there are some promising triton perf improvements in the upstream from the sterlind gptq-for-llama repo.  Any thoughts on getting sterlind bits merged into the qwopqwop repo?
